### ISSUE TITLE
agentpress init not working

### ISSUE THREAD

**Issue by JacobFV** [kortix-ai/agentpress#4]
ran this in a new project:

```zsh
agent-press-playground-py3.12jacob@Jacobs-MacBook-Pro agent-press-playground % poetry add agentpress     
Using version ^0.1.3 for agentpress

Updating dependencies
Resolving dependencies... (3.2s)

Package operations: 78 installs, 0 updates, 0 removals

  - Installing attrs (24.2.0)
  - Installing rpds-py (0.20.1)
  - Installing referencing (0.35.1)
  - Installing certifi (2024.8.30)
  - Installing charset-normalizer (3.4.0)
  - Installing h11 (0.14.0)
  - Installing idna (3.10)
  - Installing jsonschema-specifications (2024.10.1)
  - Installing markupsafe (3.0.2)
  - Installing mdurl (0.1.2)
  - Installing six (1.16.0)
  - Installing smmap (5.0.1)
  - Installing sniffio (1.3.1)
  - Installing typing-extensions (4.12.2)
  - Installing urllib3 (2.2.3)
  - Installing annotated-types (0.7.0)
  - Installing anyio (4.6.2.post1)
  - Installing filelock (3.16.1)
  - Installing frozenlist (1.5.0)
  - Installing fsspec (2024.10.0)
  - Installing gitdb (4.0.11)
  - Installing httpcore (1.0.6)
  - Installing jinja2 (3.1.4)
  - Installing jsonschema (4.23.0)
  - Installing markdown-it-py (3.0.0)
  - Installing multidict (6.1.0)
  - Installing narwhals (1.12.1)
  - Installing numpy (2.1.3)
  - Installing packaging (23.2)
  - Installing propcache (0.2.0)
  - Installing pydantic-core (2.23.4)
  - Installing pygments (2.18.0)
  - Installing python-dateutil (2.9.0.post0)
  - Installing pytz (2024.2)
  - Installing pyyaml (6.0.2)
  - Installing requests (2.32.3)
  - Installing tqdm (4.66.6)
  - Installing tzdata (2024.2)
  - Installing aiohappyeyeballs (2.4.3)
  - Installing aiosignal (1.3.1)
  - Installing altair (5.4.1)
  - Installing blinker (1.8.2)
  - Installing cachetools (5.5.0)
  - Installing click (8.1.7)
  - Installing distro (1.9.0)
  - Installing gitpython (3.1.43)
  - Installing httpx (0.27.2)
  - Installing huggingface-hub (0.26.2)
  - Installing jiter (0.7.0)
  - Installing pandas (2.2.3)
  - Installing pillow (10.4.0)
  - Installing protobuf (5.28.3)
  - Installing pyarrow (18.0.0)
  - Installing pydantic (2.9.2)
  - Installing pydeck (0.9.1)
  - Installing regex (2024.9.11)
  - Installing rich (13.9.4)
  - Installing tenacity (9.0.0)
  - Installing toml (0.10.2)
  - Installing tornado (6.4.1)
  - Installing wcwidth (0.2.13)
  - Installing yarl (1.17.1)
  - Installing zipp (3.20.2)
  - Installing aiohttp (3.10.10)
  - Installing importlib-metadata (8.5.0)
  - Installing openai (1.53.0)
  - Installing prompt-toolkit (3.0.36)
  - Installing psutil (5.9.8)
  - Installing python-dotenv (1.0.1)
  - Installing streamlit (1.39.0)
  - Installing termcolor (2.4.0)
  - Installing tiktoken (0.8.0)
  - Installing tokenizers (0.20.1)
  - Installing agentops (0.3.14)
  - Installing litellm (1.51.3)
  - Installing questionary (2.0.1)
  - Installing streamlit-quill (0.0.3)
  - Installing agentpress (0.1.3)

Writing lock file
agent-press-playground-py3.12jacob@Jacobs-MacBook-Pro agent-press-playground % poetry run agentpress init
Traceback (most recent call last):
  File "/Users/jacob/Library/Caches/pypoetry/virtualenvs/agent-press-playground-Ewpu3dAP-py3.12/bin/agentpress", line 5, in <module>
    from agentpress.cli import main
  File "/Users/jacob/Library/Caches/pypoetry/virtualenvs/agent-press-playground-Ewpu3dAP-py3.12/lib/python3.12/site-packages/agentpress/cli.py", line 7, in <module>
    import pkg_resources
ModuleNotFoundError: No module named 'pkg_resources'
```

additional info:

```
agent-press-playground-py3.12jacob@Jacobs-MacBook-Pro agent-press-playground % poetry run python --version
Python 3.12.6
agent-press-playground-py3.12jacob@Jacobs-MacBook-Pro agent-press-playground % poetry --version    
Poetry (version 1.8.3)
agent-press-playground-py3.12jacob@Jacobs-MacBook-Pro agent-press-playground % pyenv --version
pyenv 2.4.13
agent-press-playground-py3.12jacob@Jacobs-MacBook-Pro agent-press-playground % pyenv global
3.12
agent-press-playground-py3.12jacob@Jacobs-MacBook-Pro agent-press-playground % ls -a
.                       ..                      README.md               agent_press_playground  poetry.lock             pyproject.toml          tests
```

**Comment by markokraemer** [comment]
Just saw the issue, cant reproduce unfortunately :( 

Can you try with latest agentpress, just in case?

**Comment by markokraemer** [comment]
Try 'poetry add setuptools' & 'poetry run agentpress init'

**Comment by markokraemer** [comment]
Missing dependency pkg_resources which is part of the setuptools package. This is a common issue with Python 3.12 and some packages that still rely on pkg_resources

**Comment by markokraemer** [comment]
Upgrade AgentPress to latest version & it should be resolved by default as well, let me know if it is

**Comment by JacobFV** [comment]
`poetry add setuptools` fixed it! adding setuptools as a dep in #7
